File: 100-Days-Series/Day 24/PerfectNumber.js

// Arrow function for check if number is perfect number or not.
const getIsPerfectNumber=(num)=>{
    if(num<=1)
        return false;
    let sum=0;
    for(let i=1;i<=num/2;i++){
        if(num%i==0)
            sum+=i;
    }
    return sum==num;
}
// Let's Start here
let num=28;
// Call the function and display result.
if(getIsPerfectNumber(num))
    console.log(num+" is a perfect number!");
else
    console.log(num+" is not a perfect number!");

File: 100-Days-Series/Day 24/PerfectNumber.c
#include<stdio.h>
// Function for check if number perfect number or not.
int getIsPerfectNumber(int num){
    if(num<=1){
        return 0;
    }
    int sum=0;
    for(int i=1;i<=num/2;i++){
        if(num%i==0)
            sum+=i;
    }
    printf("%d",num);
    return sum==num;
}
// Let's Start here
void main(){
    int num;
    printf("\nEnter a number : ");
    scanf("%d",&num);
    // Call the function and display result.
    int res=getIsPerfectNumber(num);
    if(res)
        printf("\n%d is a perfect number!",num);
    else
        printf("\n%d is not a perfect number!",num);
}